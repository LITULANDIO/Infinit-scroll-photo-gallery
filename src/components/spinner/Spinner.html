
<transition name="fade">
    <div v-if="show" class="flex justify-center">
        <div class="h-28 |  py-5 px-5 | bottom-0 fixed flex items-center | border-2 border-yellow-600 rounded-full | text-white bg-gray-800">
            <span>Loading...</span>
        </div>
    </div>
</transition>
