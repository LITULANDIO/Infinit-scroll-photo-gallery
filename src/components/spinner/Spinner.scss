.fade-enter-from{
    opacity: 0;
}
.fade-enter-to{
    opacity: 1;
}
.fade-enter-active{
    transition: all 1s ease;
}
.fade-leave-from{
    opacity: 1;
}
.fade-leave-to{
    opacity: 0;
}
.fade-leave-active{
    transition: all 1s ease;
}

