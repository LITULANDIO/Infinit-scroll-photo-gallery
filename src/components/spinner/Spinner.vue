<template src="./Spinner.html"></template>

<script>
export default {
    name: 'Spinner',
    props:{
        show:{
            type: Boolean,
            default: false
        }
    },
    setup(){

        return{

        }

    }

}
</script>

<style lang="scss" src="./Spinner.scss" scoped></style>