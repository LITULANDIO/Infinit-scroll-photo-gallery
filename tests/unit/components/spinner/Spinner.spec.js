import { shallowMount } from '@vue/test-utils';
import Spinner from '@/components/spinner/Spinner.vue';

describe('Spinner component', () =>{

    test('must match snapshot', () =>{
        const wrapper = shallowMount(Spinner, {
            props:{
                show: false
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Verify that the spinner is initially hidden', () =>{
        const wrapper = shallowMount(Spinner, {
            props:{
                show: false
            }
        })

        const div1 = wrapper.find('div');
        expect(div1.exists(false))
    })


})
