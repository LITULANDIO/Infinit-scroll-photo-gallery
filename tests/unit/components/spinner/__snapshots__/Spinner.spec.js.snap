// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component must match snapshot 1`] = `
<transition-stub>
  <!--v-if-->
</transition-stub>
`;
